vortex-file/src/driver.rs

```diff
@@ -102,13 +98,13 @@ impl CoalescedDriver {
     /// Mark a segment as prefetched (if it hasn't been polled), and update the prefetch
     /// buffer count.
     fn mark_as_prefetched(&mut self, id: SegmentId) {
-        let state = self.segment_state(id);
-
-        // If the segment has been pre-fetched, we can remove its bytes from the buffer.
+        let state = self.segment_state_mut(id);
         assert!(!state.is_prefetched, "segment already prefetched");
+
+        // Account for these new bytes in the prefetch buffer
         if !state.polled {
+            state.is_prefetched = true;
             self.prefetched_bytes += self.segment_spec(id).length as i64;
-            self.segment_state_mut(id).is_prefetched = false;
```

Comment:
I think b/c we never were setting this true, the branch in unmark_as_prefetched below was probably never being taken